lighthouse-core/audits/seo/canonical.js

```diff
@@ -75,102 +39,102 @@ class Canonical extends Audit {
    * @param {LH.Audit.Context} context
    * @return {Promise<LH.Audit.Product>}
    */
-  static audit(artifacts, context) {
+  static async audit(artifacts, context) {
```

Comment:
perhaps you could split up this function in a few sub functions to make it more readable? A lot of things are going on it seems.

Reply:
yeah good call @wardpeet!